static/src/js/website_sale_osc.js

```diff
@@ -1,174 +1,93 @@
-(function() {
-  'use strict';
+$(document).ready(function () {
 
-  var website = openerp.website;
-
-  function startTransaction(acquirer_id, form) {
-    if (typeof openerp.website.prePaymenProcess === 'function') {
-      // the transaction process should only start when the pre payment process is
-      // finished
-      openerp.website.prePaymenProcess(function(result) {
-        openerp.jsonRpc('/shop/payment/transaction/' + acquirer_id, 'call', {}).then(function (data) {
-          form.submit();
+    function getFormData(form) {
+        var fields = form.serializeArray();
+        var form_data = {};
+        jQuery.each(fields, function (i, field) {
+            form_data[field.name] = field.value;
         });
-      });
-    } else {
-      openerp.jsonRpc('/shop/payment/transaction/' + acquirer_id, 'call', {}).then(function (data) {
-        form.submit();
-      });
+
+        return form_data;
     }
-  }
 
-  function getPostAddressFields(elms, data) {
-    elms.each(function(index) {
-      data[$(this).attr('name')] = $(this).val();
+    $("#delivery_carrier").find("input[name='delivery_type']").click(function (ev) {
+        var carrier_id = $(ev.currentTarget).val();
+        var acquirer_id = $("#payment_method").find("div.oe_sale_acquirer_button:not(:hidden)").data('id');
+        var $address_form = $("form[action='/shop/checkout']");
+        var address_data = getFormData($address_form);
+        address_data['acquirer_id'] = acquirer_id;
+        $.post("/shop/checkout", address_data, function (data) {
+            window.location.href = '/shop/payment?carrier_id=' + carrier_id;
+        })
+        .fail(function () {
+            alert("Please check again if all fields are filled out correctly");
```

Comment:
If we deactivate the "Terms & Conditions" in the configuration we get this error message. I couldn't find the correct position for this comment. I think this check was not changed and now throws this error because of the changes